pkg/printers/internalversion/printers.go

```diff
@@ -988,7 +989,7 @@ func printPod(pod *api.Pod, options printers.GenerateOptions) ([]metav1.TableRow
 
 	if pod.DeletionTimestamp != nil && pod.Status.Reason == node.NodeUnreachablePodReason {
 		reason = "Unknown"
-	} else if pod.DeletionTimestamp != nil {
+	} else if pod.DeletionTimestamp != nil && (podPhase == api.PodRunning || podPhase == api.PodPending) {
```

Comment:
nit: can we use `!util.IsPodPhaseTerminal` [here](https://github.com/kubernetes/kubernetes/blob/9e2075b3c87061d25759b0ad112266f03601afd8/pkg/api/v1/pod/util.go#L317)? This will better reflect the intention and simplify the code, but maybe there is some issue with using the package form kubectl (not sure, to be tested).